public/js/updateStudentProfile.js

```diff
@@ -0,0 +1,69 @@
+const edaidEditButton = document.querySelector('#edaidEditButton')
+const upfrontEditButton = document.querySelector('#upfrontEditButton')
+const laptopEditButton = document.querySelector('#laptopEditButton')
+const githubUsernameEditButton = document.querySelector('#githubUsernameEditButton')
+
+function handleEditClick(event) {
+    const buttonName = event.target.getAttribute('id')
+    const divName = buttonName.replace('EditButton', 'Container')
+    const containerDiv = document.querySelector('.' + divName)
+    console.log(containerDiv)
+    const fieldName = divName.replace('Container', '')
+    const description = buttonName.replace('EditButton', 'Description')
+    const descriptionTag = document.querySelector('#' + description)
+    if(divName == 'laptopContainer') {
+    containerDiv.innerHTML = 
+        '<form class="form studentProfileEditableField">' +
+        '<label>Laptop required: </label>' +
+        '<div>' +
+        '<input type="radio" value="No" id="noLaptop" name="' + fieldName + '">' +
+        '<label for="noLaptop">No</label>' +
+        '<input type="radio" value="Yes" id="yesLaptop" name="' + fieldName + '">' +
+        '<label for="yesLaptop">Yes</label>' +
+        '</div>' +
+        '<input class="saveButton btn btn-sm btn-primary" type="submit" value="Save">' +
+        '</form>'
+    } else {
+        containerDiv.innerHTML = 
+        '<form class="form studentProfileEditableField">' +
+        '<label for="' + fieldName + 'TextBox">' + descriptionTag.textContent + '</label>' +
+        '<input type="text" id="' + fieldName + 'TextBox" name="' + fieldName + '">' + 
+        '<input class="saveButton  btn btn-primary btn-sm" type="submit" value="Save">' +
+        '</form>'
+    }
+
+    const saveButton = document.querySelector('.saveButton')
+    const form = document.querySelector('.form')
+    
+    form.addEventListener('submit', (event) => {
+        event.preventDefault()
+        const data = new FormData(form, saveButton)
+        const formData = data.get(fieldName)
+        const updatedField = {
+            [fieldName]: formData,
+            id: studentId 
+        } 
+        const jsonUpdatedField = JSON.stringify(updatedField)
+        fetch('/api/updateStudentProfile', {
+            method: 'PUT',
+            body: jsonUpdatedField,
+            headers: {
+                'Content-Type': 'application/json'
+            }
+        }).then( async (response) => {
+            return response.json().then((data) => {
```

Comment:
as far as i can tell, you don't need to return here and you also don't need to make the anon arrow function async

Reply:
actually i think this whole block is a bit weird 🤔 i'll figure out shortly what i want you to do with it (probably just make it standard as per the notes)

Reply:
agree, I suggested it just to get it working but imo await is the way to go